### DISCUSSION TITLE
`pnpm ls` equivalent for Turborepo

### DISCUSSION THREAD

**Discussion by maschwenk** [vercel/turborepo#7482] [Ideas]
### Which project is this feature idea for?

Turborepo

### Describe the feature you'd like to request

`pnpm` has trouble understanding that lockfile changes can affect packages even if the package source code does not change https://github.com/pnpm/pnpm/issues/4420

From my understanding of this issue, Turborepo _can_ understand that relationship https://github.com/vercel/turbo/pull/3250

I'd like to lean on Turbo as the canonical "change detector"

Unfortunately, Turborepo requires all tasks to have a `pacakge.json` script associated with them. This would require us to either:

1. Setup a "dummy" script for each package that is a `no-op` but will allow us to navigate the DAG
2. Add a `build` script to scripts that don't really have a build step, just to ensure it's always included

### Describe the solution you'd like

More generally, it would be nice to have a way to understand Turbo's view of the DAG divorced from any particular task.

### Describe alternatives you've considered

Not sure

**Comment by ArnaudRinquin** [comment]
Wouldn't the [--dry](https://turbo.build/repo/docs/reference/command-line-reference#--dry----dry-run) argument cover your case?

**Comment by arlyon** [comment]
I have run into this as well, wanting to get information about packages in the context of the repo rather than in the context of a particular task. --dry would help if every package happens to share a common task (such as typechecking ... 😏) but I agree that assuming that wouldn't always make sense. I will feed this back and get a timeline.

Thanks for taking the time to open an issue.

**Comment by maschwenk** [comment]
> Wouldn't the [--dry](https://turbo.build/repo/docs/reference/command-line-reference#--dry----dry-run) argument cover your case?

Using `dry` requires that every package in the repo has the script in question. I'd like to use `turbo` as a replacement for my use of `pnpm ls`. As mentioned above, `pnpm` is not a clever about change detection in certain situations, whereas Turborepo can be.
